### PR TITLE
feat(viz): refactor vizconfig into initial support for render templates

### PR DESCRIPTION
**Pull request by b5**
_CI is currently failing b/c we'll also have to PR the dsdiff package, we'll have to merge dsdiff first, but I'd like to get feedback on this PR first_

@osterbit wants "qri render" templates inside of qri. After much hand-waving about security, I've thought about it, and now deeply agree that we should move to use "qri render" directly within the qri frontend to give users to express datasets however they want. So to get that to work I've done some light refactoring here, renaming to viz, and adding a slot for a single "scriptPath" file that'll be the template file.

I've removed the "Vizualizations" field that used to be in vizconfig because we don't have a use for it in this html-template direction.

The biggest change to keep an eye on is within the dsfs package, which I've adjusted to write the template file into the dataset package, following the patterns we set forth in storing transform scripts. This means the html template now travels with the dataset.

### CODE REVIEW

File: dsfs/dataset.go [hunk 1]
```diff
@@ -527,13 +527,29 @@ func WriteDataset(store cafs.Filestore, ds *dataset.Dataset, dataFile cafs.File,
 		adder.AddFile(stf)
 	}
 
-	if ds.VisConfig != nil {
-		vc, err := JSONFile(PackageFileVisConfig.String(), ds.VisConfig)
-		if err != nil {
-			return datastore.NewKey(""), fmt.Errorf("error marshaling dataset visconfig to json: %s", err.Error())
+	if ds.Viz != nil {
+		// vc, err := JSONFile(PackageFileViz.String(), ds.Viz)
```
Comment (dustmop) [inline dsfs/dataset.go]: Delete commented out code

File: dsfs/testdata/complete/template.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
```
Comment (dustmop) [inline dsfs/testdata/complete/template.html]: Is there anything that is testing the output of Render, a template that has been processed and had values rendered into it? The name "complete" implies to me that that's what this should be.

File: viz.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+// IsEmpty checks to see if Viz has any fields other than the internal path
+func (v *Viz) IsEmpty() bool {
+	return v.Format == "" && v.ScriptPath == ""
+}
+
+// SetPath sets the internal path property of a Viz
+// Use with caution. most callers should never need to call SetPath
+func (v *Viz) SetPath(path string) {
```
Comment (dustmop) [inline viz.go]: Is this only used internally? If so, can it be renamed setPath.

File: dsfs/dataset.go [hunk 4]
```diff
@@ -527,13 +527,29 @@ func WriteDataset(store cafs.Filestore, ds *dataset.Dataset, dataFile cafs.File,
...
+		// if err != nil {
+		// 	return datastore.NewKey(""), fmt.Errorf("error marshaling dataset viz to json: %s", err.Error())
+		// }
+		// fileTasks++
+		// adder.AddFile(vc)
+		if ds.Viz.Script != nil {
+			fileTasks++
+			adder.AddFile(cafs.NewMemfileReader("viz_script", ds.Viz.Script))
```
Comment (dustmop) [inline dsfs/dataset.go]: Change into a constant vizScriptTask, same as the case statement down below that looks for it. Would be nice for transform_script as well, but that can be done later.

File: viz.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+	if path == "" {
+		v.path = datastore.Key{}
+	} else {
+		v.path = datastore.NewKey(path)
+	}
+}
+
+// Assign collapses all properties of a group of structures on to one this is
+// directly inspired by Javascript's Object.assign
+func (v *Viz) Assign(visConfigs ...*Viz) {
```
Comment (dustmop) [inline viz.go]: This doesn't ever appear to use more than one parameter. Can it use non-variadic parameters?

File: viz.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+		}
+		if vs.ScriptPath != "" {
+			v.ScriptPath = vs.ScriptPath
+		}
+	}
+}
+
+// _visconfig is a private struct for marshaling into & out of.
+// fields must remain sorted in lexographical order
+type _visconfig struct {
```
Comment (dustmop) [inline viz.go]: Rename to vizconfigPod: we're no longer spelling viz with an "s", underscore unnecessary, this is playing the same role as DatasetPod, ProfilePod, etc.

File: viz_test.go [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/ipfs/go-datastore"
+	"io/ioutil"
+	"testing"
+)
+
+var Viz1 = &Viz{
```
Comment (dustmop) [inline viz_test.go]: Unexported, don't use a capital: rename these three to testViz1, testViz2, testViz3.

File: viz.go (hunk 3 above)
Comment (b5) [reply viz.go]: SetPath is used way up the stack: https://github.com/qri-io/qri/blob/abb36a425321576fe922ab72bdf2b59e8d212ce4/lib/datasets.go#L520

Once we fix that issue, we should be able to remove `SetPath` entirely, but for now, we'll need it to make an update that only affects `Viz` work properly

File: dsfs/testdata/complete/template.html (hunk 2 above)
Comment (b5) [reply dsfs/testdata/complete/template.html]: very good point. In this setting dataset has little-to-no knowledge of what to actually _do_ with the information stored in viz, so we can't test that here. I'm thinking we should rename "complete" to "all_fields", which is what I meant by "complete", and is less ambiguous.

File: viz.go (hunk 5 above)
Comment (b5) [reply viz.go]: We could, I'm hoping to deprecate `Assign` entirely. I'll open an issue to discuss, but I think we should leave the variadic-ness for now.

**Pull request merged**